test-harness/specs/parameters-ac2.oas2.txt

```diff
@@ -0,0 +1,28 @@
+====test====
+When I send a request to an operation
+And the operation has a required query param
+And in the request I sent that query param is missing
+Then I should get a 400 response back
+And the payload of the response should be in application/problem+json content-type
+====spec====
+swagger: '2.0'
+paths:
+  /pet:
+    get:
+      parameters:
+        - in: query
+          name: age
+          required: true
+          type: number
+      responses:
+        200:
+          description: ok
+          schema:
+            type: string
+====server====
+mock -p 4010
+====command====
+curl -sIXGET http://127.0.0.1:4010/pet
+====expect====
+HTTP/1.1 400 Bad Request
```

Comment:
In order to make this happen some work is going to be required @philsturgeon 

Fundamentally now our validation system returns 422 — for whatever error it encounters. We will need to pop error informations from inside instead of catching all of them and then return just 422. It should be viable in a day (or two, if something goes wrong).

The problem _could be_ the returned error message highly depends on the validation order.

I am assuming we'll do

Path Validation (Router) — Headers Validation — Body Validation.

Reply:
Thread glue: https://github.com/stoplightio/prism/pull/475#discussion_r305303199

It’s ok for now if everything is consistently a 422, I’ll accept that. I will make an issue to have certain cases be a 404